src/compile.rs

```diff
@@ -411,6 +411,11 @@ fn cargo_build_command(
         }
     }
 
+    // Set PYO3_BUILD_EXTENSION_MODULE for all PyO3 builds
+    if bridge_model.is_pyo3() {
```

Comment:
This will also set it for bin targets that use `pyo3` which are not extension modules, see
https://github.com/PyO3/maturin/blob/8ab42219247277fee513eac753a3e90e76cd46b9/src/bridge.rs#L223-L226

Reply:
thanks, fixed.